ext/standard/array.c

```diff
@@ -3228,7 +3228,7 @@ PHP_FUNCTION(array_pop)
 	}
 	ZVAL_COPY_DEREF(return_value, val);
 
-	if (!p->key && Z_ARRVAL_P(stack)->nNextFreeElement > 0 && p->h >= (zend_ulong)(Z_ARRVAL_P(stack)->nNextFreeElement - 1)) {
+	if (!p->key && p->h >= (zend_ulong)(Z_ARRVAL_P(stack)->nNextFreeElement - 1)) {
```

Comment:
That `zend_ulong` cast doesn't look right to me. Can you also add some array_pop uses to your test to check that it works correctly?

Reply:
Indeed, although it was giving the expected result, the cast was on the wrong side. I've added one example to the test.